apps/frontend/src/store/data_filters.ts

```diff
@@ -382,40 +723,226 @@ export function filter_cache_key(f: Filter) {
   return JSON.stringify(newFilter);
 }
 
+/**
+ * Filters controls by given filters
+ *
+ * @param controls - Array of contextualized controls
+ * @param filters - Filters to apply to the controls
+ * @returns Filtered array of controls
+ *
+ */
 export function filterControlsBy(
   controls: readonly ContextualizedControl[],
-  filters: Record<string, boolean | string[] | undefined>
+  filters: Record<string, FilterRecord>
 ): readonly ContextualizedControl[] {
   const activeFilters: typeof filters = _.pickBy(
     filters,
     (value, _key) =>
       (Array.isArray(value) && value.length > 0) ||
       (typeof value === 'boolean' && value)
   );
-  return controls.filter((control) => {
+
+  // Filter out specific categories
+  const firstPass = controls.filter((control) => {
+    return Object.entries(activeFilters).every(([filter, value]) => {
+      const item:
+        | SearchEntry<GenericSearchEntryValue>
+        | SearchEntry<GenericSearchEntryValue>[]
+        | boolean = _.get(control, filter);
+      if (Array.isArray(value) && typeof item !== 'boolean') {
+        return value?.some((term) => {
+          if (!term.negated) {
+            return fieldIncludes(item, (compareValue) =>
+              compareValue.toLowerCase().includes(term.value.toLowerCase())
+            );
+          } else {
+            return !fieldIncludes(item, (compareValue) =>
+              compareValue.toLowerCase().includes(term.value.toLowerCase())
+            );
+          }
+        });
+      } else {
+        return item === value;
+      }
+    });
+  });
+
+  // Overall keywords filtering
+  const final: ContextualizedControl[] = filterControlsByKeywords(firstPass);
+
+  return final;
+}
+
+/**
+ * Filters controls by keyword filters
+ *
+ * @param controls - Array of contextualized controls
+ * @returns Filtered array of controls
+ *
+ */
+export function filterControlsByKeywords(controls: ContextualizedControl[]) {
+  let results = controls;
+  if (SearchModule.keywordsSearchTerms.length > 0) {
+    for (const filter of SearchModule.keywordsSearchTerms) {
```

Comment:
I handled this by passing in the keyword search terms from the `activeFilters` variable in the main filter function for the respective type (checklist or control).  Note: Keyword filtering is explicitly ignored in the main filter function since it needs to be handled differently from the rest of the filters